src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2TimeoutTests.cs

```diff
@@ -156,13 +157,14 @@ public async Task PING_NoKeepAliveTimeout_DoesNotResetKeepAliveTimeout()
 
             CreateConnection();
 
-            await InitializeConnectionAsync(_noopApplication);
+            await InitializeConnectionAsync(_echoApplication);
 
             // Connection starts and sets keep alive timeout
             _mockTimeoutControl.Verify(c => c.SetTimeout(It.IsAny<long>(), TimeoutReason.KeepAlive), Times.Once);
             _mockTimeoutControl.Verify(c => c.ResetTimeout(It.IsAny<long>(), TimeoutReason.KeepAlive), Times.Never);
             _mockTimeoutControl.Verify(c => c.CancelTimeout(), Times.Never);
 
+            // Stream will stay open because it is waiting for request body to end
```

Comment:
Do you need to send that body eventually to unblock the app for cleanup? Or does server tear down kill the body and let it exit?

Reply:
I've always assumed that `Http2TestBase.Dispose()` cleans up any outstanding requests.

Reply:
Can you put a break point in the echo app to confirm that?

Reply:
Hmm, the connection appears to have ended but it appeared the echo app never progressed. I'll put an explicit end stream in the test.